### PR TITLE
Added app for auto demo builder using OpenAI API and Asana API

### PR DESCRIPTION
**Pull request by andrew-asana**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204507574694423

### CODE REVIEW

File: common/DemoAutoBuilder/api.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    });
+  });
+}
+
+async function handler(req, res) {
+  let session;
+  try {
+    session = await getServerSession(req, res, authOptions);
+    console.log('session: ', session);
+    console.log('access_token: ', session);
```
Comment (bengsquared) [inline common/DemoAutoBuilder/api.js:32]: remove logging of access tokens :)

File: common/DemoAutoBuilder/services/asanaService.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+                );
+              })
+            );
+          })
+        );
+      })
+    );
+    success = true;
+  } catch (error) {
+    /* TODO: Handle error */
```
Comment (bengsquared) [inline common/DemoAutoBuilder/services/asanaService.js:195]: I guess we just gotta make sure these are a fast follow lol

**Review by bengsquared — APPROVED** [review]
LGTM! some small fixes, should be quick

File: common/DemoAutoBuilder/controllers/autoBuilderController.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    of "sections" in each of these projects. Make each of the sections of the ${projectType} style workflow. \
+    Assume all projects will follow the same structure. Please also generate example tasks, the amount of tasks in each section is \
+    at your discretion, for each based off the theme of the sections. Generate at least 3 or more tasks per section in each project. \
+    Generate at least 2 or more projects per portfolio.  Generate your answer as neatly formed JSON that can be consumed by another platform.
+    Use the following JSON structure: ${jsonSchema}.
+    Simulate the "due_date"(s) for each task starting in this current date ${currentDate}. \
+    Do not generate other text in your response aside from the JSON object itself. 
+  `;
+
+  const response = await submitOpenAiPrompt(promptMessage);
```
[the remark this answers is not in the document]
Comment (andrew-asana) [reply common/DemoAutoBuilder/controllers/autoBuilderController.js:63]: Definitely will explore it. Next iteration will most likely spawn multiple calls to OpenAI.

File: common/DemoAutoBuilder/services/asanaService.js (hunk 2 above)
Comment (andrew-asana) [reply common/DemoAutoBuilder/services/asanaService.js:195]: I'll add something for now in the interm.

File: common/DemoAutoBuilder/api.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+async function handler(req, res) {
+  let session;
+  try {
+    session = await getServerSession(req, res, authOptions);
+    console.log('session: ', session);
+  } catch (error) {
+    console.log('error: ', error);
+  }
+
+  await applyCors(req, res, cors);
```
Comment (bengsquared) [inline common/DemoAutoBuilder/api.js:36]: can remove CORS

**Review by bengsquared — APPROVED** [review]

**Pull request merged**